editor/src/core/workers/parser-printer/parser-printer-comments.ts

```diff
@@ -0,0 +1,64 @@
+import * as TS from 'typescript'
+import { Comment, multiLineComment, singleLineComment } from '../../shared/element-template'
+import { fastForEach } from '../../shared/utils'
+
+export interface CommentsResult {
+  leadingComments: Array<Comment>
+}
+
+export function getComments(sourceText: string, node: TS.Node): CommentsResult {
+  let leadingComments: Array<Comment> = []
+  TS.forEachLeadingCommentRange(
+    sourceText,
+    node.getFullStart(),
+    (pos: number, end: number, commentKind: TS.CommentKind, hasTrailingNewLine: boolean) => {
+      switch (commentKind) {
+        case TS.SyntaxKind.SingleLineCommentTrivia:
```

Comment:
"Trivia"? Ha!

Reply:
Trivia is what the TypeScript compiler treats anything that could be elided, so comments and whitespace for example.